google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/gcs/GcsToolWindowAction.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.gcs;
+
+import com.google.cloud.tools.intellij.ui.GoogleCloudToolsIcons;
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.DumbAwareAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.wm.ToolWindow;
+import com.intellij.openapi.wm.ToolWindowManager;
+
+/**
+ * Creates a shortcut action to open the Google Cloud storage explorer in the Google Cloud Tools
+ * tool menu group.
+ */
+public final class GcsToolWindowAction extends DumbAwareAction {
+
+  public GcsToolWindowAction() {
+    super(
+        GctBundle.message("gcs.tools.menu.action.text"),
+        GctBundle.message("gcs.tools.menu.action.description"),
+        GoogleCloudToolsIcons.CLOUD_STORAGE);
+  }
+
+  @Override
+  public void actionPerformed(AnActionEvent event) {
+    Project project = event.getProject();
+    if (project != null) {
+      ToolWindow window =
+          ToolWindowManager.getInstance(project).getToolWindow("Google Cloud Storage");
```

Comment:
Could `.getToolWindow()` ever reasonably return `null`? The Javadoc says it can but I'm wondering in what real-world scenario this could happen.

Reply:
It won't return null since we register a tool window of this ID in plugin.xml. But its conceivable that another plugin (or some bad code in ours) could unregister this tool window, but I don't think we need to handle this realistically.

Reply:
Ok cool, that's what I figured.